core/http-auth-aws-crt/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/DefaultAwsCrtV4aHttpSigner.java

```diff
@@ -0,0 +1,105 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.crt.internal.signer;
+
+import static software.amazon.awssdk.http.auth.aws.crt.internal.CrtUtils.toCredentials;
+import static software.amazon.awssdk.http.auth.aws.util.SignerConstant.PRESIGN_URL_MAX_EXPIRATION_DURATION;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.crt.auth.signing.AwsSigningConfig;
+import software.amazon.awssdk.http.auth.aws.crt.AwsCrtV4aHttpSigner;
+import software.amazon.awssdk.http.auth.spi.AsyncSignRequest;
+import software.amazon.awssdk.http.auth.spi.AsyncSignedRequest;
+import software.amazon.awssdk.http.auth.spi.SignRequest;
+import software.amazon.awssdk.http.auth.spi.SyncSignRequest;
+import software.amazon.awssdk.http.auth.spi.SyncSignedRequest;
+import software.amazon.awssdk.identity.spi.AwsCredentialsIdentity;
+
+/**
+ * An implementation of a {@link AwsCrtV4aHttpSigner} that uses properties to compose v4a-signers in order to delegate signing of
+ * a request and payload (if applicable) accordingly.
+ */
+@SdkInternalApi
+public final class DefaultAwsCrtV4aHttpSigner implements AwsCrtV4aHttpSigner {
+
+    /**
+     * Given a request with a set of properties, determine which signer to delegate to, and call it with the request.
+     */
+    private static AwsCrtV4aHttpSigner getDelegate(
+        SignRequest<?, ? extends AwsCredentialsIdentity> signRequest) {
+        String regionName = signRequest.requireProperty(REGION_NAME);
+        String serviceSigningName = signRequest.requireProperty(SERVICE_SIGNING_NAME);
+        Clock signingClock = signRequest.requireProperty(SIGNING_CLOCK, Clock.systemUTC());
+        boolean doubleUrlEncode = signRequest.requireProperty(DOUBLE_URL_ENCODE, true);
+        boolean normalizePath = signRequest.requireProperty(NORMALIZE_PATH, true);
+        AuthLocation authLocation = signRequest.requireProperty(AUTH_LOCATION, AuthLocation.HEADER);
+        Duration expirationDuration = validateExpirationDuration(
+            signRequest.requireProperty(EXPIRATION_DURATION, PRESIGN_URL_MAX_EXPIRATION_DURATION)
+        );
+        boolean isPayloadSigning = signRequest.requireProperty(PAYLOAD_SIGNING_ENABLED, true);
+
+        AwsSigningConfig signingConfig = new AwsSigningConfig();
+        signingConfig.setCredentials(toCredentials(signRequest.identity()));
+        signingConfig.setService(serviceSigningName);
+        signingConfig.setRegion(regionName);
+        signingConfig.setAlgorithm(AwsSigningConfig.AwsSigningAlgorithm.SIGV4_ASYMMETRIC);
+        signingConfig.setTime(signingClock.instant().toEpochMilli());
+        signingConfig.setUseDoubleUriEncode(doubleUrlEncode);
+        signingConfig.setShouldNormalizeUriPath(normalizePath);
```

Comment:
Can we extract this logic to a separate method to make this method smaller, something like `signingConfig(xxx)` (forgot to leave a comment in the previous review)?